quilt/tools/command.py

```diff
@@ -326,10 +328,45 @@ def _clone_git_repo(url, branch, dest):
     cmd += [url, dest]
     subprocess.check_call(cmd)
 
+def _log(**kwargs):
+    # TODO(dima): Save logs to a file, then send them when we get a chance.
+
+    cfg = _load_config()
+    if cfg.get('disable_analytics'):
+        return
+
+    session = _get_session()
+
+    # Disable error handling.
+    session.hooks.update(dict(
+        response=None
+    ))
+
+    try:
+        session.post(
+            "{url}/api/log".format(
+                url=get_registry_url(),
+            ),
+            data=json.dumps([kwargs]),
+            timeout=LOG_TIMEOUT,
+        )
+    except requests.exceptions.RequestException:
+        # Ignore logging errors.
+        pass
+
 def build(package, path=None, dry_run=False, env='default'):
     """
     Compile a Quilt data package, either from a build file or an existing package node.
     """
+    path_str = str(path) if path else None
+    try:
+        _build_internal(package, path, dry_run, env)
+    except Exception as ex:
+        _log(type='build', package=package, path=path_str, dry_run=dry_run, env=env, error=str(ex))
+        raise
+    _log(type='build', package=package, path=path_str, dry_run=dry_run, env=env)
```

Comment:
Package names are potentially very sensitive (sometimes even more than the actual data). We should obscure the real package names and paths at least for private packages. We should probably obscure paths or not send them even for public packages.

Reply:
Local path we can leave out but package names need to be in there for Mixpanel events to work properly. In fact the events should support the same fields that current events do. Keep in mind that all traffic is https. If there is still some exposure of the names we can issue a public key for the clients but yes that's more complicated.

Reply:
Keep in mind that these are packages that haven't been pushed to Quilt so they might be private packages that users never intend to push. The names and contents of those packages haven't been shared with us so we shouldn't look at them. I think we can safely send package names for packages that have already been pushed.

Reply:
Too complicated to track whether or not package has been pushed (requires statefulness to know if package has been pushed) and doesn't solve core problem of diagnosing build failures. We don't touch contents at any point in the logs. Considering we have zero build analytics right now we shouldn't get stuck on trying to micro-optimize this. Everything is encrypted in transit, never publicly logged, and users can actually disable all logging with an environment variable.

Reply:
So here's what we can do:
* Cryptographically hash package names
* Don't send build path

Reply:
I'm also logging exceptions - and those could contain paths or package names. Do we want to change anything there?